gnovm/pkg/gnolang/uverse.go

```diff
@@ -1003,3 +1003,27 @@ func copyNativeToData(dst []byte, rv reflect.Value, rvl int) {
 		dst[i] = uint8(rv.Index(i).Uint())
 	}
 }
+
+const FormatUndefined = "undefined"
+
+// formatTypeValue formats the given TypedValue based on its type and returns the formatted string representation.
+func formatTypeValue(ev *TypedValue, m *Machine) string {
```

Comment:
What's the advantage of doing this in a separate function versus making the changes inside `TypeValue.Sprint`? Part of this implementation duplicates what is already there.

Reply:
I actually didn't realize that the function and `TypeValue.Sprint` had the same functionality at that moment. I've now changed it to use `Sprint`, and the part of the test where the string was being printed twice has also been fixed. Thank you for pointing that out 👍

Reply:
I'm not sure, but I'm guessing it's because the `formatTypeValue` method doesn't handling some features properly. I think that integrating this `formatTypeValue` function into `TypeValue.Sprint` might solve it.

I also think the additional points you made below are related too.

Reply:
Actually, I didn’t realize that it could be merged with TypeValue.Sprint. However, after some modifications, it's now incorporated into the internal logic of TypeValue.Sprint. 

By doing so, the println now satisfies both the test cases from before the modification and the new tests as well. thanks for pointing out 👍